indico/modules/designer/placeholders.py

```diff
@@ -407,3 +408,65 @@ class FixedImagePlaceholder(DesignerPlaceholder):
     def render(cls, item):
         buf = DesignerImageFile.get(item['image_id']).open()
         return Image.open(buf)
+
+
+class RegistrationFormFieldPlaceholder(DesignerPlaceholder):
+    """Placeholder representing a `RegistrationData` instance for a given regform field.
+
+    Unlike other placeholders which are always present, registration data depends
+    on the linked regform and thus the placeholders must be generated on the fly.
+    """
+
+    group = 'regform_fields'
+
+    def __init__(self, field):
+        self.field = field
+        self.name = f'field-{field.id}'
+
+    @property
+    def description(self):
+        return self.field.title
+
+    def render(self, registration):
+        data_by_field = registration.data_by_field
+        data = data_by_field.get(self.field.id, None)
+        if data is None:
+            return ''
+        return self._render(data)
+
+    def _render(self, data):
+        friendly_data = data.friendly_data
+        if friendly_data is None:
+            return ''
+
+        if self.field.input_type == 'multi_choice':
+            return ' · '.join(friendly_data)
+        elif self.field.input_type == 'accommodation':
+            return self._render_accommodation(friendly_data)
+        else:
+            return friendly_data
+
+    def _render_accommodation(self, friendly_data):
+        if friendly_data['is_no_accommodation']:
+            return _('No accommodation')
+
+        arrival = format_date(friendly_data['arrival_date'])  # TODO: include locale
+        departure = format_date(friendly_data['departure_date'])
+        accommodation = friendly_data['choice']
+        return _('''\
+Arrival: {arrival} · Departure {departure} · Accommodation: {accommodation}
+''').format(arrival=arrival, departure=departure, accommodation=accommodation)
```

Comment:
I'd rather use a neutral version that does not require i18n, e.g. `{accommodation} ({arrival} - {departure})`.
When someone really puts that field on a badge, they'll probably have to put static text around it anyway.

Because if no event language is set (few events do that) (and even then you'd need to force the event locale which I don't think we do anywhere when printing badges?), using the language of the person generating the badge - in particular if it's a manager - is not ideal. Image a Polish or Chinese user at CERN printing badges for some event happening here... ;)